apps/rush-lib/src/logic/PublishGit.ts

```diff
@@ -44,13 +44,13 @@ export class PublishGit {
       workingDirectory ? workingDirectory : process.cwd());
   }
 
-  public addTag(shouldExecute: boolean, packageName: string, packageVersion: string): void {
+  public addTag(shouldExecute: boolean, packageName: string, packageVersion: string, commitId: string | undefined): void {
     // Tagging only happens if we're publishing to real NPM and committing to git.
     const tagName: string = PublishUtilities.createTagname(packageName, packageVersion);
     PublishUtilities.execCommand(
       !!this._targetBranch && shouldExecute,
       'git',
-      ['tag', '-a', tagName, '-m', `${packageName} v${packageVersion}`]);
+      ['tag', '-a', tagName, '-m', `"${packageName} v${packageVersion}"`, ...(commitId ? [commitId] : [])]);
```

Comment:
Why are the quotes needed?

Reply:
Because there is a space separating the packageName and packageVersion. If there's no quotes, the command won't work adding a commit id on the end.

It parsed fine before because it was the last parameter.

eg `git tag -a foo_v0.0.1 -m "foo v0.0.1" abc123` vs `git tag -a foo_v0.0.1 -m oo v0.0.1 abc123`

Reply:
But we're explicitly passing the arguments separately, so the shell shouldn't need to parse them.

Unless `PublishUtilities` is doing something silly.

Reply:
oh, huh... the IIRC, the dryrun output didn't add the quotes around it

Reply:
Yeah, here's the dryrun output (with quotes).

```
* DRYRUN: .../web-build-tools/common/temp/pnpm-local/node_modules/.bin/pnpm pack (stack/rush-stack-compiler-3.3)

* DRYRUN: git tag -a @microsoft/rush-stack-compiler-3.3_v0.4.10 -m "@microsoft/rush-stack-compiler-3.3 v0.4.10" abc123 
```

Reply:
I mean, it makes sense, how would `PublishUtilities` know to put quotes around something or not?

Reply:
Displaying command-line arguments is actually a somewhat common need.  Perhaps we should add a helper to the [Executable](https://github.com/microsoft/rushstack/blob/5627415032539c27e93fdddac4edf93d2c87a311/libraries/node-core-library/src/Executable.ts#L113) API, something like this:

```ts
/**
 * Formats {@link Executable.spawnSync} style arguments to be displayed.
 * Arguments containing spaces or special characters will be properly
 * escaped in shell quotes.
 * 
 * @remarks
 * This method is intended for display purposes only. It is not guaranteed
 * to produce a command string that will execute correctly on every shell. 
 */
public static formatArgs(filename: string, args: string[]): string {
    let result: string = filename;
    for (const arg of args) {
        if (result.length > 0) {
            result += ' ';
        }
        if (/[\s\""]/.test(arg)) {
            result += JSON.stringify(arg);
        } else {
            result += arg;
        }
    }
    return result;
}
```